pkg/operator/controller/externaldns/deployment.go

```diff
@@ -105,8 +114,18 @@ func (r *reconciler) currentExternalDNSDeployment(ctx context.Context, nsName ty
 	return true, depl, nil
 }
 
+// extractCredentialsSecret retieves the contents of the credentials' secret
```

Comment:
As the function only has one caller maybe it is simpler and more straightforward to just have this code inline at the call site. It would remove the hard part (i.e., naming!).

Reply:
Inlined the function as suggested by @frobware .